CMakeLists.txt

```diff
@@ -594,6 +594,30 @@ include_directories(SYSTEM ${PROJECT_BINARY_DIR}/include)
 
 find_package(Threads REQUIRED)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+  file(
+    WRITE "${CMAKE_BINARY_DIR}/librarycheck.cc"
+    "int main() { return 0; }")
+
+  execute_process(
+    COMMAND "${CMAKE_CXX_COMPILER}" "-###" "-std=c++17" "librarycheck.cc"
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    OUTPUT_VARIABLE HUGE_COMPILE_OUTPUT
+    ERROR_VARIABLE HUGE_COMPILE_OUTPUT
+    )
+
+  if(HUGE_COMPILE_OUTPUT MATCHES "\"-lstdc\\+\\+\"")
+    set(CLANG_CXXLIB "libstdc++")
```

Comment:
nit, the variable name is sort of misleading. as it has `clang` in it. how about `CXX_STDLIB`?